### PR TITLE
Add configMap as a volume type

### PR DESCRIPTION
**Pull request by tengqm**
Closes: #2385

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6766)

### CODE REVIEW

File: docs/concepts/storage/volumes.md [hunk 1]
```diff
@@ -171,6 +172,43 @@ writers simultaneously.
...
+
+The [`configMap`](/docs/tasks/configure-pod-container/configmap/) resource
+provides a way to inject configuration data into Pods.
+The data stored in a `ConfigMap` object can be referenced in a `configMap` type
+of volume and then consumed by containerized applications running in a Pod.
+
+When referencing a `configMap` object, you can simply provide its name in the
+volume to reference it. You can also customize the path to use for a specific
+entry in the ConfigMap.
+For example:
```
Comment (abiogenesis-now) [inline docs/concepts/storage/volumes.md]: Replace "For example:" with:
>  For example, to mount the `log-config` ConfigMap onto a Pod called`configmap-pod`, you might use the YAML below:

This makes it clearer to the reader what they're about to see.

File: docs/concepts/storage/volumes.md [hunk 2]
```diff
@@ -171,6 +172,43 @@ writers simultaneously.
...
+  volumes:
+    - name: config-vol
+      configMap:
+        name: log-config
+        items:
+          - key: log_level
+            path: log_level
+```
+
+The Pod created using the above manifest references a ConfigMap object in the
```
Comment (abiogenesis-now) [inline docs/concepts/storage/volumes.md]: If you make the change suggested above, you can probably take out the first sentence and instead have something like:

> The `log-config` ConfigMap is mounted as a volume, and all contents stored in its `log_level` entry is mounted into the Pod at path `/etc/config/log_level`. (Note that this path is derived from the volume's `mountPath` and the `path` keyed with `log_level`).

File: docs/concepts/storage/volumes.md [hunk 3]
```diff
@@ -171,6 +172,43 @@ writers simultaneously.
 
 See the [CephFS example](https://github.com/kubernetes/examples/tree/{{page.githubbranch}}/staging/volumes/cephfs/) for more details.
 
+### configMap
+
+The [`configMap`](/docs/tasks/configure-pod-container/configmap/) resource
+provides a way to inject configuration data into Pods.
+The data stored in a `ConfigMap` object can be referenced in a `configMap` type
```
[the remark this answers is not in the document]
Comment (tengqm) [reply docs/concepts/storage/volumes.md]: ok

File: docs/concepts/storage/volumes.md (hunk 2 above)
Comment (tengqm) [reply docs/concepts/storage/volumes.md]: sounds good

**Comment by abiogenesis-now** [comment]
thanks for the change!
/lgtm

**Pull request merged**